lib/src/soloud.dart

```diff
@@ -628,6 +629,38 @@ interface class SoLoud {
     return completer.future;
   }
 
+  /// Stops the engine and disposes of all resources, including sounds
+  /// and the audio isolate in an synchronous way.
+  /// 
+  /// This method is meant to be called when exiting the app. For example
+  /// within the `dispose()` of the uppermost widget in the tree
+  /// or inside [AppLifecycleListener.onExitRequested].
+  /// 
+  /// During the normal app life cycle and if you want to shutdown the player,
+  /// please use [shutdown] which safer and it is meant to throw errors.
+  void deinit() {
+    _log.finest('deinit() called');
+    /// check if we are in the middle of an initialization.
+    if (_initializeCompleter != null) {
+      _log.warning('deinit() called while already initializing.');
+      _initializeCompleter = null;
+    }
+
+    /// if not already initialized, just return.
+    if (!_isInitialized) {
+      // The engine isn't initialized.
+      _log.warning('deinit() called when the engine is not initialized');
+      return;
+    }
+
+    /// reset broadcast and kill isolate
+    _cleanUpUnsuccessfulInitialization();
```

Comment:
I'd rather spell this out (inline). That method is really meant only to clean up after unsuccessful initialization.